Source/Extensions/Blazorise.Charts.Zoom/ChartZoom.razor.cs

```diff
@@ -16,23 +17,139 @@ public partial class ChartZoom<TItem> : ChartPlugin<TItem, JSChartZoomModule>
 
     /// <inheritdoc/>
     protected override JSChartZoomModule CreatePluginJsModule()
-        => new( JSRuntime, VersionProvider, BlazoriseOptions );
+    {
+        DotNetObjectRef ??= DotNetObjectReference.Create( new ChartZoomAdapter<TItem>( this ) );
+
+        return new( JSRuntime, VersionProvider, BlazoriseOptions );
+    }
 
     /// <inheritdoc/>
     protected override async Task InitializePlugin()
-        => await JSModule.AddZoom( ParentChart.ElementId, Options );
+        => await JSModule.Initialize( DotNetObjectRef, ParentChart.ElementId, Options );
 
     /// <inheritdoc/>
     protected override bool UpdatePluginParameters( ParameterView parameters )
         => parameters.TryGetValue<ChartZoomPluginOptions>( nameof( Options ), out var paramOptions ) && !Options.IsEqual( paramOptions );
 
+    /// <inheritdoc/>
+    protected override ValueTask DisposeAsync( bool disposing )
+    {
+        if ( disposing && Rendered )
+        {
+            if ( DotNetObjectRef != null )
+            {
+                DotNetObjectRef.Dispose();
+                DotNetObjectRef = null;
+            }
+        }
+
+        return base.DisposeAsync( disposing );
+    }
+
+    /// <summary>
+    /// Asynchronously retrieves the current zoom level of the chart.
+    /// </summary>
+    /// <remarks>The zoom level represents the scaling factor applied to the chart. A value of <see
+    /// langword="1.0"/>  indicates no zoom, while values greater than <see langword="1.0"/> indicate zooming in, and
+    /// values  less than <see langword="1.0"/> indicate zooming out.</remarks>
+    /// <returns>A <see cref="ValueTask{TResult}"/> representing the asynchronous operation.  The result is a <see
+    /// cref="double"/> value indicating the current zoom level of the chart.  The default value is <see
+    /// langword="1.0"/> if the JavaScript module is not initialized.</returns>
+    public async ValueTask<double> GetZoomLevel()
+    {
+        if ( JSModule is null )
```

Comment:
The same - is there a normal workflow state where the `JSModule` would be null? Or is it exceptional state?